pkg/cri/resource-manager/control/rdt/rdt.go

```diff
@@ -223,6 +240,11 @@ func rdtError(format string, args ...interface{}) error {
 // Register us as a controller.
 func init() {
 	control.Register(RDTController, "RDT controller", getRDTController())
-	config.GetModule(rdt.ConfigModuleName).AddNotify(getRDTController().configNotify)
+	config.Register(ConfigModuleName, "RDT control", getRDTController().opt, getRDTController().defaultOptions)
+	config.GetModule(ConfigModuleName).AddNotify(getRDTController().configNotify)
```

Comment:
nitpick: you can specify the notification callback already during registration like this:
```
config.Register(ConfigModuleName, "RDT control", getRDTController().opt, getRDTController().defaultOptions,
        config.WithNotify(getRDTController().configNotify))
``` 
I think this should be the preferred way of registering a callback within the same package, unless you have to use a method of an object which has not been instantiated yet at the time of registration (which, for instance, is usually the case for policies).

Reply:
Ach, I totally messed up with the different versions of the PR 😕 Now, I think I managed to push the right version which actually works.

In this, we really need to separate `Register()` and `AddNotify()`. We cannot add the notifier before the RDT adaptation (goresctrl) has been initialized and is ready to accept config changes.